runtime/src/main/java/io/micronaut/scheduling/NextFireTime.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.scheduling;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.function.Supplier;
+
+import io.micronaut.scheduling.cron.CronExpression;
+
+/**
+ * Represents the next fire time for a cron expression.
+ * @author croudet
+ */
+class NextFireTime implements Supplier<Duration> {
```

Comment:
I know this has already been merged, but this should include `@since` and be marked as `@Internal`. Would make the class final too.